api/list_cluster.go

```diff
@@ -26,57 +27,29 @@ func ListCluster(w http.ResponseWriter, req *http.Request) {
 	var err error
 	config.Cfg.UserID, err = config.ValidateUser(authHeader, apiKeyHeader)
 	if err != nil {
-		log.Printf(err.Error())
+		log.Println("ERROR: validating user: ", err)
 		http.Error(w, "error validating user", http.StatusUnauthorized)
 		return
 	}
-	dbPath := config.Cfg.Common.ProjectDir + "/" + config.Cfg.UserID
-	clusterInfos := ReadClusterInfo(dbPath, config.Cfg.UserID)
-	clusterByte, err := json.Marshal(clusterInfos)
+	path := config.Cfg.Common.ProjectDir + "/" + config.Cfg.UserID + "/" + config.Cfg.UserID + ".db"
+	db, err := metastore.NewDb(path)
```

Comment:
I was maybe wrong. Initializing db for every `REST` call seems off.

I don't know whether we can have many sqlite instance of same db. If it doesn't support, then multiple async calls will break this. 

If having many db open is the concern, we can have some kind of garbage collection where it'll close the db instance when it's reached much threshold. 

Simple thing would be having a mapping of tenant to db instance in a hashmap and won't worry about having many db instances. 

can be optimised when we have more tenant (Cynical me pops everywhere)

Reply:
Yeah, agree. Same pattern being done in other functions too. Will think about how to not do that.

simple way I can think of is to define a package level variable and initialize it. Not sure that’s preferred way.

Reply:
Like this https://github.com/spinup-host/spinup/pull/98/files#diff-d715f391f2ffa038c3167bfea1adf741c1f5e39f08349fbbacab780a594146a4R41

Reply:
Maybe you can have all the handlers as methods instead of functions. 

see whether this works https://github.com/poonai/inspektor/blob/main/controlplane/handlers/datasource.go#L47